### PR TITLE
SER Support

### PR DESCRIPTION
**Pull request by JaiOCP**
SER Support is added on top of base infra introduced by
https://github.com/opencomputeproject/SAI/pull/1777

### CODE REVIEW

File: inc/saitypes.h [hunk 1]
```diff
@@ -1722,14 +1722,141 @@ typedef enum _sai_object_stage_t
...
+    uint32_t ser_log_type;
+} sai_ser_health_data_t;
+
+/**
+ * @extraparam sai_health_data_type_t data_type
+ */
+typedef union _sai_health_data_t
+{
+    /** @validonly data_type == SAI_HEALTH_DATA_TYPE_SER */
+    sai_ser_health_data_t ser;
```
Comment (kcudnik) [inline inc/saitypes.h:1850]: Why there is only one field where? Dont create stricte with one wntry

**Comment by kcudnik** [comment]
Fix build errors

File: inc/saitypes.h (hunk 1 above)
Comment (itaibaz) [reply inc/saitypes.h:1850]: This is the original design discussed in 1777 so seems good
Per health event we create a struct
Each health event is only of one type at a time, so we use a union

So let's say we have 2 types of event
struct health_a_data
struct health_b_data

we will put in the union
union health_data {a_data, b_data}

**Comment by JaiOCP** [comment]
Opened a new PR
https://github.com/opencomputeproject/SAI/pull/1982

**Pull request closed without merging**